ambry-rest/src/main/java/com.github.ambry.rest/Http2ChannelPoolMap.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.rest;
+
+import com.github.ambry.config.SSLConfig;
+import com.github.ambry.network.RequestInfo;
+import io.netty.bootstrap.Bootstrap;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.pool.AbstractChannelPoolMap;
+import io.netty.channel.pool.ChannelPool;
+import io.netty.channel.socket.nio.NioSocketChannel;
+
+
+public class Http2ChannelPoolMap extends AbstractChannelPoolMap<RequestInfo, ChannelPool> {
+  private final EventLoopGroup eventLoopGroup;
+  private final SSLConfig sslConfig;
+
+  public Http2ChannelPoolMap(SSLConfig sslConfig, EventLoopGroup eventLoopGroup) {
```

Comment:
Maybe construct `SSLFactory` outside (i.e. in the `NetworkClientFactory` impl) and pass it into this constructor since it should be able to shared across all instances.

Reply:
SSLFactory is used.